app/src/main/java/com/kevo/displaydemo/ui/secondarydisplay/SimplePresentationFragment.kt

```diff
@@ -48,20 +68,90 @@ class SimplePresentationFragment(
             }
         }
 
+        setupViewPager()
+
         return binding.root
     }
 
+    override fun onResume() {
+        super.onResume()
+        startOrResetTimer()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        timer?.cancel()
+    }
+
     override fun onDestroyView() {
         super.onDestroyView()
         _binding = null
     }
 
+    private fun setupViewPager() = with(binding) {
+        viewPager = this.infiniteViewPager
+
+        val sliderImages = mutableListOf<SlideItem>()
+        sliderImages.add(
+            SlideItem(Key.CustomerOrderEntry, R.drawable.example_screen_customer_order_total)
+        )
+        sliderImages.add(
+            SlideItem(Key.LoyaltyPhone, R.drawable.example_screen_loyalty_phone)
+        )
+        sliderImages.add(
+            SlideItem(Key.ReceiptSelection, R.drawable.example_screen_receipt_selection)
+        )
+        sliderImages.add(
+            SlideItem(Key.SampleAd4, R.drawable.sample_ad_4)
+        )
+
+        slidingImageAdapter = SlidingImageAdapter(sliderImages, viewPager)
+        viewPager.adapter = slidingImageAdapter
+
+        viewPager.clipToPadding = false
+        viewPager.clipChildren = false
+        viewPager.offscreenPageLimit = 3
+        viewPager.getChildAt(0).overScrollMode = RecyclerView.OVER_SCROLL_NEVER
+        viewPager.registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
+            override fun onPageSelected(position: Int) {
+                val item: SlideItem? = slidingImageAdapter.getItemAt(position)
+                // Useful for debugging
+                // Log.v(TAG, "Now showing item: ${item?.key}, ${item?.imageResId}")
+                when (item?.key) {
+                    Key.CustomerOrderEntry -> btnContainer.isVisible = true
+
+                    else -> btnContainer.isVisible = false
+                }
+            }
+        })
+    }
+
     fun setText(newText: String) {
         binding.secondScreenTextView.text = newText
     }
 
+    private fun startOrResetTimer() {
+        Log.d(TAG, "Resetting scheduled timer")
+
+        timer?.cancel()
+        timer = Timer().apply {
+            this.scheduleAtFixedRate(object : TimerTask() {
+                override fun run() {
+                    foregroundEvents.launch { transitionToNextSlide() }
+                }
+            }, AUTOMATIC_SLIDE_INTERVAL, AUTOMATIC_SLIDE_INTERVAL)
+        }
+    }
+
+    @MainThread
+    private fun transitionToNextSlide() {
+        Log.d(TAG, "Sliding to item ... ${viewPager.currentItem + 1}")
+        viewPager.currentItem = viewPager.currentItem + 1
+    }
+
     companion object {
 
-        const val TAG = "SimplePresentationFragment"
+        const val TAG = "SimplePresentFragment"
```

Comment:
The Code Analyzer was complaining that this tag was 26 chars long, and the `android.util.Log` CPP implementation only allows for 23 chars